code-reviews/recipes/Terraform.md

```diff
@@ -53,10 +63,19 @@ echo -e "\n\n>>> Terraform validate"
 terraform validate
 ```
 
-> TODO: Add example of how to use this in an AzDO build validation pipeline
-
 ## Code Review Checklist
 
-In addition to the [Code Review Checklist](../readme.md) you should also look for these terraform specific code review items
-
-> TODO: Add checklist
+In addition to the [Code Review Checklist](../readme.md) you should also look for these Terraform specific code review items
+
+* [ ] Are all providers used in the terraform scripts [versioned](https://www.terraform.io/docs/configuration/providers.html#provider-versions) to prevent breaking changes in the future?
+* [ ] Are modules split into separate .tf files where appropriate?
+* [ ] Does the repository contain a `README.md` describing the architecture provisioned?
+* [ ] Is the Terraform project configured using Azure Storage as remote state backend?
+* [ ] Is the remote state backend storage account key is stored a secure location (e.g. Azure Key Vault)?
+* [ ] If Terraform code is mixed with application source code, is the Terraform code isolated into a dedicated folder?
+* [ ] If the infrastructure going to different depending on the environment (e.g. Dev, UAT, Production), does the environment specific parameters are supplied via .tfvars file?
+* [ ] Are the resource definition and data sources are handled correctly in the Terraform?
+    resource : Indicates to Terraform that the current configuration is in charge of managing the life cycle of the object
+    data: Indicates to Terraform that you only want to get a reference of the existing object, but don’t want to manage it part of this configuration
+* [ ] Does the code is split into multiple reusable modules?
+* [ ] Are unit tests used for Terraform code (e.g. [`Terratest`](https://terratest.gruntwork.io/))?, While running the unit tests from DevOps pipelines, check for managed identities or service principal are used
```

Comment:
"While running the unit tests from DevOps pipelines, check for managed identities or service principal are used"

What does this mean? should managed identities or service principals be used for unit tests? if so, 
-> Verify that managed identities or service principals are used to run the tests

If not:
-> Verify that managed identities or service principals are not used to run the tests

Reply:
Yes, managed identities or service principals should be used, I will update accordingly.

Reply:
@TessFerrandez - In fact this statement is not needed, as it is anyways a DevOps requirement and not just for Terraform.

Reply:
reworded based on the review comment.